core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowSnapshotResource.scala

```diff
@@ -0,0 +1,113 @@
+package edu.uci.ics.texera.web.resource.dashboard.workflow
+
+import edu.uci.ics.texera.web.SqlServer
+import org.glassfish.jersey.media.multipart.FormDataParam
+import javax.annotation.security.PermitAll
+import io.dropwizard.auth.Auth
+import org.jooq.types.UInteger
+import edu.uci.ics.texera.web.auth.SessionUser
+import javax.ws.rs.core.MediaType
+import javax.ws.rs._
+import org.jooq.Condition
+import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
+import edu.uci.ics.texera.web.model.jooq.generated.Tables.WORKFLOW_SNAPSHOT
+import edu.uci.ics.texera.web.resource.dashboard.workflow.WorkflowAccessResource.WorkflowAccess
+import edu.uci.ics.texera.web.model.jooq.generated.tables.daos.WorkflowSnapshotDao
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.WorkflowSnapshot
+import edu.uci.ics.texera.web.resource.dashboard.workflow.WorkflowSnapshotResource._
+
+object WorkflowSnapshotResource {
+  final private lazy val context = SqlServer.createDSLContext()
+  private val snapshotDao = new WorkflowSnapshotDao(context.configuration)
+
+  /**
+    *  This function get snapshot by the sid
+    */
+  def getSnapshotbyId(sid: UInteger): WorkflowSnapshot = {
+    snapshotDao.fetchOneBySid(sid)
+  }
+
+  /**
+    * This function retrieves the latest snapshot of a workflow
+    * @return sid
+    */
+  def getLatestSnapshot(): UInteger = {
```

Comment:
this is incorrect, you are assuming these snapshots are all for the same workflow, what if the latest, is for a different workflow

Reply:
add wid column into workflow snapshot table